test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1010 @@
+
+	clientConfig, err := loader.ClientConfig()
+	if err != nil {
+		t.Fatalf("error geting client config: %#v", err)
+	}
+	client, err := newClient(*clientConfig)
+	if err != nil {
+		t.Fatalf("error creating client: %#v", err)
+	}
+
+	if _, err := kubeClient.Core().Namespaces().Create(&kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: testNamespace}}); err != nil {
+		t.Fatalf("error creating test namespace: %#v", err)
+	}
+
+	gvkSeen := map[unversioned.GroupVersionKind]empty{}
+	gvrSeen := map[unversioned.GroupVersionResource]empty{}
+
+	for gvk, apiType := range kapi.Scheme.AllKnownTypes() {
+		// we do not care about internal objects or lists // TODO make sure this is always true
+		if gvk.Version == runtime.APIVersionInternal || strings.HasSuffix(apiType.Name(), "List") {
+			continue
+		}
+
+		kind := gvk.Kind
+		pkgPath := apiType.PkgPath()
+
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			gvkSeen[gvk] = empty{}
+			_, ok := gvkWhiteList[gvk]
+			if ok {
+				t.Logf("skipping test for %s from %s because its GVK %s is whitelisted and has no mapping", kind, pkgPath, gvk)
+			} else {
+				t.Errorf("no mapping found for %s from %s but its GVK %s is not whitelisted", kind, pkgPath, gvk)
+			}
+			continue
+		}
+
+		gvr := gvk.GroupVersion().WithResource(mapping.Resource)
+		gvrSeen[gvr] = empty{}
+
+		ptrType := reflect.PtrTo(apiType)
+		testData, ok := etcdStorageData[gvr]
+
+		if !ok {
+			t.Errorf("no test data for %s from %s.  Please add a test for your new type to etcdStorageData.", kind, pkgPath)
+			continue
+		}
+
+		if testData.ephemeral { // TODO it would be nice if we could remove this and infer if an object is not stored in etcd
+			t.Logf("Skipping test for %s from %s", kind, pkgPath)
+			continue
+		}
+
+		if isInKindAndPathWhiteList(kind, pkgPath) {
+			t.Logf("kind and path are whitelisted: skipping test for %s from %s", kind, pkgPath)
+			continue
+		}
+
+		if len(testData.expectedEtcdPath) == 0 || len(testData.stub) == 0 {
+			t.Errorf("empty test data for %s from %s", kind, pkgPath)
+			continue
+		}
+
+		obj, err := jsonToObject(testData.stub, gvk)
+		if err != nil || reflect.TypeOf(obj) != ptrType || isZero(reflect.ValueOf(obj)) {
+			t.Errorf("invalid test data for %s from %s: %s", kind, pkgPath, err)
+			continue
+		}
+
+		func() { // forces defer to run per iteration of the for loop
+			all := &[]cleanupData{}
+			defer func() {
+				if !t.Failed() { // do not cleanup if test has already failed since we may need things in the etcd dump
+					if err := client.cleanup(all); err != nil {
+						t.Fatalf("failed to clean up etcd: %#v", err)
+					}
+				}
+			}()
+
+			if err := client.createPrerequisites(mapper, testNamespace, testData.prerequisites, all); err != nil {
+				t.Errorf("failed to create prerequisites for %s from %s: %#v", kind, pkgPath, err)
+				return
+			}
+
+			if !isInCreateAndCompareWhiteList(obj) { // do not try to create whitelisted items
```

Comment:
wouldn't need a whitelist, you'd just check the given path and expect an object of the given type